Libs/DICOM/Widgets/ctkDICOMBrowser.cpp

```diff
@@ -1092,10 +1092,22 @@ bool ctkDICOMBrowser::confirmDeleteSelectedUIDs(QStringList uids)
   ctkMessageBox confirmDeleteDialog;
   QString message = tr("Do you want to delete the following selected items?");
 
+  // calculate maximum number of rows that fit in the browser widget to have a reasonable limit
+  // on the items to show in the dialog
+  int browserHeight = this->geometry().height();
+  int patientsTableRowHeight = d->dicomTableManager->patientsTable()->tableView()->rowHeight(0);
+  int maxNumberOfPatientsToShow = browserHeight / patientsTableRowHeight - 1; // Subtract 1 due to the do not show checkbox
```

Comment:
If it is safer then we could subtract 2-3. Also make sure that the value is at least 3 (or so).

Reply:
I just didn't want to use arbitrary numbers (1 seemed easy to justify with the checkbox that is within the same layout). With the status bar and the toolbar and window header and OS taskbar even 1 is more than safe.

But I can make it 3, it's safer for sure. And I'll add the minimum number.